### PR TITLE
update dev workflow

### PR DESCRIPTION
**Pull request by dkarnutsch**
- add pre-commit hook
- remove docker-compose, use nvm instead
- format project-root with prettier

### CODE REVIEW

**Review by nsams — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -1,20 +1,19 @@
...
-    docker-compose -f docker-compose.yml -f docker-compose.mac.yml up # on OSX
+Run `install.sh` once.
 
-### use yarn
+### Run build and storybook
 
-    docker-compose exec comet-admin bash -c "yarn --help"
-    docker-compose exec comet-admin bash -c "cd packages/admin && yarn --help"
+```
+npm run start
```
Comment (johnnyomair) [inline README.md:14]: Shouldn't this be `yarn start`? Or does `lerna run ...` use yarn under the hood anyway? As far as I know, we need to use yarn in combination with lerna.

Comment (nsams) [reply README.md:14]: btw: do we still need yarn - as a recent npm relase added support for workspaces (afaik compatible with yarn)

the whole nvm setup does somehow ignore yarn anyway :)

Comment (johnnyomair) [reply README.md:14]: Found out that lerna uses yarn as configured in `lerna.json`, so please ignore my remark. 😅

Comment (dkarnutsch) [reply README.md]: should indeed be `yarn ...`, fixed now

**Review by johnnyomair — APPROVED** [review]

**Review by fraxachun — APPROVED** [review]

**Review by rabengraph — APPROVED** [review]

File: .nvmrc [hunk 2]
```diff
@@ -0,0 +1 @@
+14
```
Comment (kaufmo) [inline .nvmrc:1]: when we beginn to use nvm here then we have to use yarn as npm package and not as global binary anymore

File: .travis.yml [hunk 3]
```diff
@@ -1,17 +1,22 @@
...
     - sudo apt update && sudo apt install yarn
 
 jobs:
     include:
-        - name: "Linting"
+        - name: "Linting Project Root"
           script:
-              - INSTALL_DEPS=1 RUN_AS_UID=`id -u` ./lint.sh
+              - yarn install
+              - npx prettier -c .
```
Comment (kaufmo) [inline .travis.yml:15]: I think when we use yarn we should also run every cmd with yarn

File: lint-staged.config.js [hunk 4]
```diff
@@ -0,0 +1,4 @@
+module.exports = {
+    "packages/**/*.{ts,tsx,js,jsx,json,css,scss,md}": "npm run lint",
```
Comment (kaufmo) [inline lint-staged.config.js:2]: same here

**Review by kaufmo — COMMENTED** [review]
when we beginn to use nvm here we have to add yarn as npm dependency

**Comment by dkarnutsch** [comment]
@nsams @fraxachun @kaufmo @rabengraph @johnnyomair I have re-requested your reviews, as since the last review we have switched from yarn to npm. react-color versions are pinned now, because an update breaks the build.

File: README.md (hunk 1 above)
Comment (johnnyomair) [inline README.md:14]:
```suggestion
npm start
```

**Review by johnnyomair — APPROVED** [review]

**Review by nsams — APPROVED** [review]

**Review by rabengraph — APPROVED** [review]

**Review by kaufmo — APPROVED** [review]

**Pull request merged**